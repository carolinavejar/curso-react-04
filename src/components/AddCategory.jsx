import { useState } from "react"
import PropTypes from 'prop-types'

export const AddCategory = ( { onNewCategory } ) => {

    const [inputValue, setInputValue] = useState("");

    const onInputChanged = ( {target} ) => {
        setInputValue(target.value);
    }

    const onSubmit = ( evt ) => {
        console.log("======== on submit ==========");
        evt.preventDefault();
        const newInputValue = inputValue.trim();
        
        if ( newInputValue.length <= 1 ) return;
        onNewCategory(newInputValue)
        setInputValue('');
    }

    return (
        <form onSubmit= { onSubmit } aria-label="form">
            <input 
            type = "text"
            placeholder = "Buscar Gifs"
            value = { inputValue }
            onChange = { onInputChanged } />
        </form>
        
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}
