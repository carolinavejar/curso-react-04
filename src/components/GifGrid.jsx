import PropTypes from 'prop-types';
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifItem } from "./GifItem";

export const GifGrid =  ( { category } ) => {
    const { images, isLoading } = useFetchGifs(category);

    return (
        <>
            <h4> { category } </h4>
            {
                isLoading && (<h4> Cargando ... </h4>)
            }
            
            <div className="card-grid">
            { 
                images.map((image)=>  
                (
                    <GifItem key = { image.id }
                    { ...image }
                    ></GifItem>
                )
            )}
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}