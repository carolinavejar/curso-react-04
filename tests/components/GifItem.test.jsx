import { render } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas en GifItem', ()=> {
    test('Debe hacer mach con el snapshot', ()=> {
        const title = "Titulo ...";
        const url = "http://";
        const { container } = render
        (< GifItem title= {title} url={ url} />);

        expect (container).toMatchSnapshot();
    });
})