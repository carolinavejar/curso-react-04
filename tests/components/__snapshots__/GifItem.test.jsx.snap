// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem Debe hacer mach con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Titulo ..."
      src="http://"
    />
  </div>
</div>
`;

exports[`test prueba Debe hacer mach con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Titulo ..."
      src="Titulo ..."
    />
  </div>
</div>
`;
